packages/core/src/dashboards/component/AOPChart/AOPChart.tsx

```diff
@@ -0,0 +1,158 @@
+ */
+
+import React, {useEffect, useState} from 'react';
+import {Dimensions, View} from 'react-native';
+import {Chart, BarChart, LineChart, PieChart} from '@axelor/aos-mobile-ui';
+import {
+  fetchActionView,
+  fetchChartDataset,
+  fetchTypeChart,
+  getChartParameter,
+} from '../../../api/chart-api';
+
+interface AOPChartProps {
+  actionViewName: string;
+}
+
+const AOPChart = ({actionViewName}: AOPChartProps) => {
+  const [chart, setChart] = useState({type: '', dataset: [], title: ''});
+
+  useEffect(() => {
+    const fetchChartData = async () => {
+      const result = {type: '', dataset: [], title: ''};
+
+      try {
+        const actionViewResponse = await fetchActionView({
+          actionViewName: actionViewName,
+        });
+
+        const context = actionViewResponse?.data?.data[0]?.view?.context;
+        const chartName = actionViewResponse?.data?.data[0]?.view.views[0].name;
+
+        const typeResponse = await fetchTypeChart({chartName: chartName});
+        result.title = typeResponse?.data?.data?.title;
+        result.type = typeResponse?.data?.data?.series[0].type;
+        const onInit = typeResponse?.data?.data?.onInit;
+
+        let parameter = null;
+        if (onInit) {
+          const paramResponse = await getChartParameter({
+            chartName,
+            action: onInit,
+            context: context,
+          });
+          parameter = paramResponse?.data?.data[0].values;
+        }
+
+        const datasetResponse = await fetchChartDataset({
+          chartName,
+          parameter,
+          context,
+        });
+        result.dataset = datasetResponse?.data?.data?.dataset;
+      } catch (error) {
+        setChart({type: '', dataset: [], title: ''});
+      }
+
+      setChart(result);
+    };
+
+    fetchChartData();
+  }, [actionViewName]);
+
+  const BarChartRender = (datasets, title) => {
+    return (
+      <BarChart
+        datasets={datasets}
+        widthGraph={Dimensions.get('window').width}
+        title={title}
+      />
+    );
+  };
+
+  const LineChartRender = (datasets, title) => {
+    return (
+      <LineChart
+        datasets={datasets}
+        widthGraph={Dimensions.get('window').width}
+        title={title}
+      />
+    );
+  };
+
+  const PieChartRender = (datasets, type, title) => {
+    return (
+      <PieChart
+        datasets={datasets[0]}
+        widthGraph={Dimensions.get('window').width}
+        donut={type === Chart.chartType.donut}
+        title={title}
+      />
+    );
+  };
+
+  const renderChar = (type, dataset, title) => {
+    switch (type) {
+      case Chart.chartType.bar:
+        return BarChartRender(dataset, title);
+
+      case Chart.chartType.pie:
+      case Chart.chartType.donut:
+        return PieChartRender(dataset, type, title);
+
+      case Chart.chartType.line:
+        return LineChartRender(dataset, title);
+
+      default:
+        return null;
+    }
+  };
```

Comment:
Maybe we can create a ChartComponent in UI which will do the render switch to avoid doing it on Dashboard component and AOP component.